db/document.go

```diff
@@ -167,29 +167,31 @@ func NewDocument(docid string) *Document {
 
 // Accessors for document properties.  To support lazy unmarshalling of document contents, all access should be done through accessors
 func (doc *Document) Body() Body {
-	if doc._body == nil && doc.rawBody != nil {
-		err := doc._body.Unmarshal(doc.rawBody)
+	if doc._body == nil && doc._rawBody != nil {
+		err := doc._body.Unmarshal(doc._rawBody)
 		if err != nil {
 			base.Warnf(base.KeyAll, "Unable to unmarshal document body from raw body : %s", err)
 			return nil
 		}
-		doc.rawBody = nil
 	}
 	return doc._body
 }
 
 func (doc *Document) RemoveBody() {
 	doc._body = nil
-	doc.rawBody = nil
+	doc._rawBody = nil
 }
 
-// TODO: review whether this can just return raw body when available
-func (doc *Document) MarshalBody() ([]byte, error) {
-	marshalled, err := json.Marshal(doc.Body())
-	if err != nil {
-		return []byte{}, pkgerrors.Wrapf(err, "Error marshalling JSON")
+func (doc *Document) BodyBytes() ([]byte, error) {
+	if doc._rawBody == nil && doc._body != nil {
+		bodyBytes, err := json.Marshal(doc._body)
+		if err != nil {
+			err = pkgerrors.Wrapf(err, "Error marshalingl document body bytes")
+			return nil, err
```

Comment:
```suggestion
			return nil, pkgerrors.Wrapf(err, "Error marshalling document body")
```
Can probably streamline this code as the above.